addons/bus/models/bus.py

```diff
@@ -129,6 +129,10 @@ def notify():
     def _sendone(self, channel, notification_type, message):
         self._sendmany([[channel, notification_type, message]])
 
+    @api.model
+    def _senduser(self, notification_type, message):
```

Comment:
Hello I like the idea, but this name doesn't read really well. Some ideas below.

`_senduser` "Sending a user" doesn't really make a lot of sense.
`_send_to_user`? Which user?
`_to_self`?

Method `_sendone` (or similar) on `res.users`/`res.partner`?
`self.env.user._send(type, payload)` would be an actual shortcut.

Idea would be to have a similar method on other objects to which we send bus notifications. channel, guest, group, etc

What about `_sendmany`? Batching should be the preference. Isn't it annoying to have a completely different syntax for both?

cc @tsm-odoo

Reply:
What are you gaining ? Just avoid giving the first argument which is the record being notified ? I find the old syntax clearer (with_user makes no sense, it is used to browse something as a user, not target a user), and bloating models with "_send_bus" similar methods just to send notifications seems useless.

Reply:
No security review because of potential leakage of the channel

Reply:
> What about `_sendmany`? Batching should be the preference. Isn't it annoying to have a completely different syntax for both?

If we implemented it on channels, guests, groups, etc like you suggested, we could do:
```python
users._send_bus(...)
channels._send_bus(...)
```

And this would do a `_sendmany` for all users/channels for instance.

But I'm not sure if it is safe for all those models? I suppose it is, and a random user cannot listen to any arbitrary channel :laughing:

Reply:
Then remove the "send_user" or "with_user" naming, and go for something more model_based. Adding it on bus model is misleading: the base model is user, not bus who should be globally agnostic on what kind of record he is running on. It is a technical model.

So combining everything would lead to
```suggestion
    class ResUser():

        def _bus_sendone(self, notification_type, content):
            self.ensure_one()
            return self.env["bus.bus"]._send_one(self.partner_id.id, notification_type, content)
```

To match classic python guidelines, it begins with ``_bus`` as it is a "mixin-like" method, allowing to namespace them (then have _bus_sendmulti or stuff like that when we need it).

We could think of having a method on BaseModel and just define the channel getter on models, could be interesting but not sure it is worth it. But don't define "send_user", "send_channel", ... on bus, the right base model on which is should be defined is the one defining the channel key, aka user or channel, not bus.

Cheers !